### PR TITLE
Transcribe attributes from span links to events

### PR DESCRIPTION
**Pull request by seh**
Links attached to OpenTelemetry spans carry attributes like spans and events. Since we publish links as Honeycomb events, and events all carry fields, transcribe attributes on links as fields on the corresponding Honeycomb events.

Include OpenTelemetry resource attributes as the underlay, and shadow those resource attributes if necessary with the link-level attributes, as we did for spans and span events in #71.

### CODE REVIEW

**Comment by seh** [comment]
Note that this patch is stacked on top of #95, but GitHub won't allow me to represent that properly here.

**Comment by MikeGoldsmith** [comment]
@seh I've merged #95 but it looks like this PR has got a little knotted up. Please could you rebase?

Thanks!

**Comment by seh** [comment]
> Please could you rebase?

Yes, it looks cleaner now. Please take another look.

**Comment by asdvalenzuela** [comment]
@seh Apologies for the hold up with this PR. If you would please rebase again I will make sure this gets reviewed promptly. 🙏

**Comment by seh** [comment]
> If you would please rebase again I will make sure this gets reviewed promptly.

This was a blast from the past. Please see the latest commit.

**Review by asdvalenzuela — COMMENTED** [review]
One question for clarification, otherwise looks good! Thanks!

File: honeycomb/honeycomb.go [hunk 1]
```diff
@@ -496,35 +503,29 @@ func (e *Exporter) exportSpan(ctx context.Context, data *trace.SpanData) {
...
+		// Treat resource-defined attributes as underlays, with any same-keyed message event
+		// attributes taking precedence. Apply them first.
+		applyResourceAttributes(ev)
+		transcribeAttributesTo(ev, attrs)
+	}
 
 	// Treat resource-defined attributes as underlays, with any same-keyed span attributes taking
 	// precedence. Apply them first.
 	applyResourceAttributes(ev)
-	if len(e.serviceName) != 0 {
```
[the remark this answers is not in the document]
Comment (seh) [reply honeycomb/honeycomb.go:508]: See lines 508-511 above. It still happens as part of the call to `applyResourceAttributes` at line 521.

**Review by asdvalenzuela — APPROVED** [review]

**Pull request merged**